Resources/config/services.xml

```diff
@@ -4,47 +4,47 @@
         </service>
 
-        <service id="httplug.auto_discovery.auto_discovered_client" class="Http\Client\HttpClient">
+        <service id="httplug.auto_discovery.auto_discovered_client" class="Http\Client\HttpClient" public="true">
             <factory class="Http\Discovery\HttpClientDiscovery" method="find" />
         </service>
 
-        <service id="httplug.auto_discovery.auto_discovered_async" class="Http\Client\HttpAsyncClient">
+        <service id="httplug.auto_discovery.auto_discovered_async" class="Http\Client\HttpAsyncClient" public="true">
             <factory class="Http\Discovery\HttpAsyncClientDiscovery" method="find" />
         </service>
 
         <!-- Discovery with autowiring support for Symfony 3.3+ -->
-        <service id="httplug.message_factory.default" class="Http\Message\MessageFactory">
+        <service id="httplug.message_factory.default" class="Http\Message\MessageFactory" public="true">
             <factory class="Http\Discovery\MessageFactoryDiscovery" method="find" />
         </service>
         <service id="Http\Message\MessageFactory" alias="httplug.message_factory" public="false" />
         <service id="Http\Message\RequestFactory" alias="httplug.message_factory" public="false" />
         <service id="Http\Message\ResponseFactory" alias="httplug.message_factory" public="false" />
 
-        <service id="httplug.stream_factory.default" class="Http\Message\StreamFactory">
+        <service id="httplug.stream_factory.default" class="Http\Message\StreamFactory" public="true">
             <factory class="Http\Discovery\StreamFactoryDiscovery" method="find" />
         </service>
         <service id="Http\Message\StreamFactory" alias="httplug.stream_factory" public="false" />
 
-        <service id="httplug.uri_factory.default" class="Http\Message\UriFactory">
+        <service id="httplug.uri_factory.default" class="Http\Message\UriFactory" public="true">
             <factory class="Http\Discovery\UriFactoryDiscovery" method="find" />
         </service>
         <service id="Http\Message\UriFactory" alias="httplug.uri_factory" public="false" />
 
-        <service id="httplug.async_client.default" class="Http\Client\HttpAsyncClient">
+        <service id="httplug.async_client.default" class="Http\Client\HttpAsyncClient" public="true">
             <factory class="Http\Discovery\HttpAsyncClientDiscovery" method="find" />
         </service>
         <service id="Http\Client\HttpAsyncClient" alias="httplug.async_client.default" public="false" />
 
-        <service id="httplug.client.default" class="Http\Client\HttpClient">
+        <service id="httplug.client.default" class="Http\Client\HttpClient" public="true">
             <factory class="Http\Discovery\HttpClientDiscovery" method="find" />
         </service>
-        <service id="Http\Client\HttpClient" alias="httplug.client" public="false" />
+        <service id="Http\Client\HttpClient" alias="httplug.client" public="true" />
```

Comment:
hm, is this what we need to make public? this is just the alias to enable autowiring. this is not defining httplug.client

Reply:
I agree. These aliases shouldn't be public (in Symfony, they are private too).